datajunction-server/datajunction_server/construction/build.py

```diff
@@ -1094,3 +1156,207 @@ def build_ast(  # pylint: disable=too-many-arguments
     end = time.time()
     _logger.info("Finished building query in %s", end - start)
     return query
+
+
+def metrics_to_measures(
+    session: Session,
+    metric_nodes: List[Node],
+) -> DefaultDict[str, Set[str]]:
+    """
+    Returns a mapping between each metric node and a list of necessary measures for that metric.
```

Comment:
If we solidify the concepts of measures and dimension attributes as semantic types then this will likely become `metrics_to_semantic_types()`.